engine/framework/registry/factory.go

```diff
@@ -81,7 +87,15 @@ func (f *SimpleWorkerFactory[T, C]) DeserializeConfig(configBytes []byte) (Worke
 	var config C
 	config = reflect.New(reflect.TypeOf(config).Elem()).Interface().(C)
 	if err := json.Unmarshal(configBytes, config); err != nil {
-		return nil, errors.Trace(err)
+		return nil, errors.Annotate(errDeserializeConfig, err.Error())
 	}
 	return config, nil
 }
+
+// IsRetryableError implements WorkerFactory.IsRetryableError
+func (f *SimpleWorkerFactory[T, C]) IsRetryableError(err error) bool {
+	if libErrors.Is(err, errDeserializeConfig) {
```

Comment:
can the error from all implementations of `DeserializeConfig` become non-retryable?

Reply:
It depends how `DeserializeConfig` is implemented, whether it contains retryable operation such as network access etc.